source/runtime.hpp

```diff
@@ -221,10 +225,14 @@ namespace reshade
 		unsigned int _vertices = 0;
 		unsigned int _drawcalls = 0;
 
-		std::vector<effect> _effects;
+		std::vector<effect> _effects, _previous_effects;
+		std::unordered_map<size_t, effect> _recent_effects;
 		std::vector<texture> _textures;
 		std::vector<technique> _techniques;
 
+		// === Runtime Description ===
+		std::string _renderer_name;
```

Comment:
I'd just use the `_renderer_id` for the cache file names instead of this, to avoid the new field. The cache file names are already not very humand-readable anyway (because of the hash).

Reply:
Don't know if you saw my comment "Hi, I'm completely new to github, but I would like to help with this fascinating project. I have been using ReShade for maybe a week now and making preset mods for ReShade on nexus mods. Is there any way I could help, or a more appropriate place to ask these questions? Again, totally new to github haha"